### PR TITLE
Move the call to FullscreenMessageDelegate.shouldShowMessage off the ui thread

### PR DESCRIPTION
**Pull request by rymorale**
## Description
Resolves https://github.com/adobe/aepsdk-core-android/issues/347.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java [hunk 1]
```diff
@@ -70,6 +72,7 @@ class AEPMessage implements FullscreenMessage {
     private Animation dismissAnimation;
     private Animation.AnimationListener animationListener;
     private Map<String, String> assetMap = Collections.emptyMap();
+    private final Executor executor;
```
Comment (praveek) [inline code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:75]: Can you avoid having an executor inside each AEPMessage instance? You can have a single lazy initialized executor in Android UI Service and pass it to the constructor.

File: code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java [hunk 2]
```diff
@@ -127,24 +131,10 @@ class AEPMessage implements FullscreenMessage {
...
-                    ServiceConstants.LOG_TAG,
-                    TAG,
-                    "Message couldn't be displayed, FullscreenMessageDelegate#shouldShowMessage"
-                            + " states the message should not be displayed.");
-            return;
-        }
+        show(true);
+    }
 
+    public void show(final boolean withDelegateControl) {
```
Comment (praveek) [inline code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:137]: Is this method meant to be public? If so, it is missing in the FullScreenMessage interface.

File: code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java [hunk 3]
```diff
@@ -186,55 +176,74 @@ public void show() {
...
-
-                        if (currentMessageFragment != null) {
-                            fragmentManager
-                                    .beginTransaction()
-                                    .remove(currentMessageFragment)
-                                    .commit();
-                        }
+                () -> {
+                    // add the frame layout to be replaced with the message fragment
+                    rootViewGroup.addView(fragmentFrameLayout);
```
Comment (praveek) [inline code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:185]: Can you check `messagesMonitor.show(message, delegateControl)` first and run all the UI logic later?

**Review by praveek — CHANGES_REQUESTED** [review]
@rymorale I changed the base branch to dev-v2.0.0. Please resolve the conflicts.

File: code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java (hunk 2 above)
Comment (rymorale) [reply code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:137]: my mistake, this shouldn't be public.

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java [hunk 4]
```diff
@@ -127,24 +133,10 @@ class AEPMessage implements FullscreenMessage {
...
-                    ServiceConstants.LOG_TAG,
-                    TAG,
-                    "Message couldn't be displayed, FullscreenMessageDelegate#shouldShowMessage"
-                            + " states the message should not be displayed.");
-            return;
-        }
+        show(true);
+    }
 
+    void show(final boolean withDelegateControl) {
```
Comment (praveek) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:139]: Do we have a use case where we might call `show(false)`?

Comment (rymorale) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:139]: `show(true)` will be called when the Messaging extension creates an AEPMessage object then tries to show it. This call occurs within the Messaging extension. `show(false)` is called from within a customer's FullscreenMessageDelegate implementation. A customer can store the AEPMessage and decide to show it at a later time by calling AEPMessage.show().

Comment (praveek) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:139]: AEPMessage class is not public. If customers need to call show(false) then the method should be added to https://github.com/adobe/aepsdk-core-android/blob/dev-v2.0.0/code/core/src/main/java/com/adobe/marketing/mobile/services/ui/FullscreenMessage.java#L18

**Review by praveek — CHANGES_REQUESTED** [review]

**Review by praveek — APPROVED** [review]

**Pull request merged**